src/catalog/catalog.cpp

```diff
@@ -137,7 +137,20 @@ CatalogEntry *Catalog::GetEntry(ClientContext &context, CatalogType type, string
 		schema_name = DEFAULT_SCHEMA;
 	}
 	auto schema = GetSchema(context, schema_name);
-	return schema->GetEntry(context, type, name, if_exists);
+
+	if(type != CatalogType::SCALAR_FUNCTION) {
+		return schema->GetEntry(context, type, name, if_exists);
+	}
+	else {
+		//First look up at the main context schema
+		auto entry = schema->GetEntry(context, type, name, true);
+		//Case the function wans't found, it might be an UDF function, then check up the temporary schema
+		if(!entry) {
+			schema = GetSchema(context, TEMP_SCHEMA);
```

Comment:
This already happens right above it if the schema was not specified (i.e. schema_name = INVALID_SCHEMA). It shouldn't be repeated here. Instead, what I think should happen is that UDF function calls should be instantiated with INVALID_SCHEMA instead of DEFAULT_SCHEMA (e.g. see function_expression.cpp:21).